tools/gcs-bucket-mover/gcs_bucket_mover/bucket_mover_service.py

```diff
@@ -0,0 +1,657 @@
+        kms_key_name: The name of the KMS key that the project should be given access to
+    """
+
+    kms_client = discovery.build(
+        'cloudkms', 'v1', credentials=config.source_project_credentials)
+
+    # Get the current IAM policy and add the new member to it.
+    crypto_keys = kms_client.projects().locations().keyRings().cryptoKeys()  # pylint: disable=no-member
+    policy_request = crypto_keys.getIamPolicy(resource=kms_key_name)
+    policy_response = policy_request.execute(num_retries=5)
+    bindings = []
+    if 'bindings' in policy_response.keys():
+        bindings = policy_response['bindings']
+    members = []
+    members.append('serviceAccount:' + config.target_service_account_email)
+    new_binding = dict()
+    new_binding['role'] = 'roles/cloudkms.cryptoKeyEncrypterDecrypter'
+    new_binding['members'] = members
+    bindings.append(new_binding)
+    policy_response['bindings'] = bindings
+
+    # Set the new IAM Policy.
+    request = crypto_keys.setIamPolicy(
+        resource=kms_key_name, body={'policy': policy_response})
+    request.execute(num_retries=5)
+
+    spinner.write('{} {} added as Enrypter/Decrypter to key: {}'.format(
+        CHECKMARK, config.target_service_account_email, kms_key_name))
+
+
+def _assign_target_project_to_topic(spinner, config, topic_name, topic_project):
+    """Gives the service_account_email the Publisher role for the topic.
+
+    Args:
+        spinner: The spinner displayed in the console
+        config: A Configuration object with all of the config values needed for the script to run
+        topic_name: The name of the topic that the target project should be assigned to
+        topic_project: The name of the project that the topic belongs to
+    """
+
+    client = pubsub.PublisherClient(
+        credentials=config.source_project_credentials)
+    topic_path = client.topic_path(topic_project, topic_name)  # pylint: disable=no-member
+    policy = client.get_iam_policy(topic_path)  # pylint: disable=no-member
+
+    policy.bindings.add(
+        role='roles/pubsub.publisher',
+        members=['serviceAccount:' + config.target_service_account_email])
+
+    policy = client.set_iam_policy(topic_path, policy)  # pylint: disable=no-member
+
+    spinner.write('{} {} added as a Publisher to topic: {}'.format(
+        CHECKMARK, config.target_service_account_email, topic_name))
+
+
+def _run_and_wait_for_sts_job(sts_client, target_project, source_bucket_name,
+                              sink_bucket_name):
+    """Kick off the STS job and wait for it to complete. Retry if it fails.
+
+    Args:
+        sts_client: The STS client object to be used
+        target_project: The name of the target project where the STS job will be created
+        source_bucket_name: The name of the bucket where the STS job will transfer from
+        sink_bucket_name: The name of the bucket where the STS job will transfer to
+    """
+
+    counter = 0
```

Comment:
You're already importing and using the retries module, why implement your own retry logic here?

Reply:
This is a case of multiple iterations of review/refactor on this and the previous method that added the retrying module. I've updated this one retrying as well.